File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/DogeCVTest.java
package org.firstinspires.ftc.teamcode;

import com.qualcomm.robotcore.eventloop.opmode.OpMode;
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;

import org.openftc.easyopencv.OpenCvCamera;
import org.openftc.easyopencv.OpenCvCameraRotation;
import org.openftc.easyopencv.OpenCvInternalCamera;

@TeleOp
public class DogeCVTest extends OpMode {
    private OpenCvCamera phoneCam;
    private final SkystoneDetector skystoneDetector = new SkystoneDetector(telemetry);

    @Override
    public void init() {
        int cameraViewId = hardwareMap.appContext.getResources().getIdentifier(
                "cameraMonitorViewId", "id",
                hardwareMap.appContext.getPackageName());
        phoneCam = new OpenCvInternalCamera(OpenCvInternalCamera.CameraDirection.BACK, cameraViewId);
        phoneCam.openCameraDevice();
        phoneCam.setPipeline(skystoneDetector);
        phoneCam.startStreaming(960, 720, OpenCvCameraRotation.UPRIGHT);
    }

    @Override
    public void loop() {
        int position = skystoneDetector.skystonePosition();
        telemetry.addData("Skystone positions", position);
        telemetry.update();
    }

    @Override
    public void stop() {
        phoneCam.stopStreaming();
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/DogeCVTest.kt
package org.firstinspires.ftc.teamcode

import com.qualcomm.robotcore.eventloop.opmode.OpMode
import com.qualcomm.robotcore.eventloop.opmode.TeleOp

import org.openftc.easyopencv.OpenCvCamera
import org.openftc.easyopencv.OpenCvCameraRotation
import org.openftc.easyopencv.OpenCvInternalCamera

@TeleOp
class DogeCVTest : OpMode() {
    private var phoneCam: OpenCvCamera? = null
    private val skystoneDetector = SkystoneDetector(telemetry)

    override fun init() {
        val cameraViewId = hardwareMap.appContext.resources.getIdentifier(
                "cameraMonitorViewId", "id",
                hardwareMap.appContext.packageName)
        phoneCam = OpenCvInternalCamera(OpenCvInternalCamera.CameraDirection.BACK, cameraViewId)
        phoneCam!!.openCameraDevice()
        phoneCam!!.setPipeline(skystoneDetector)
        phoneCam!!.startStreaming(960, 720, OpenCvCameraRotation.UPRIGHT)
    }

    override fun loop() {
        val positions = skystoneDetector.skystonePositions()
        telemetry.addData("Skystone positions", positions)
        telemetry.update()
    }

    override fun stop() {
        phoneCam!!.stopStreaming()
    }
}
